### DISCUSSION TITLE
Deploying Gateway Controller with v0.17.0

### DISCUSSION THREAD

**Discussion by orrymr** [argoproj/argo-events#1126] [Q&A] [answered]
I am currently using an old version of Argo Events (v0.10.0). I'd like to upgrade to the latest stable version. The problem is that I have Gateways installed on my cluster. I'd like to follow [this guide](https://github.com/argoproj/argo-events/wiki/Migration-path-for-v0.17.0) in order to upgrade.
However, I am stuck on the first point:
`Deploy the gateway-controller with v0.17.0 which is configured to delete existing gateway deployments and services.`

How can I do this?

Thanks!

**Comment by whynowy** [comment, marked as answer]
I would recommend to clean up everything and install the latest version since you are running a very old version. If the service is very critical which can not afford a long downtime, probably using a different cluster and migrating the services one by one is better if that's possible.

**Reply by orrymr** [reply]
Thanks!

The Argo CRD's I currently have installed on the cluster are: 

- Gateways (gateways.argoproj.io)
- Sensors (sensors.argoproj.io)
- Workflows (workflows.argoproj.io)

So - I should be able to uninstall old Gateways & Sensors without it affecting Workflows?

Further - just to be clear, I'd need to uninstall both the CRDs and operators associated with Argo Events?

**Reply by whynowy** [reply]
Uninstall those CRDs will not impact workflows.

You can delete the controller deployments, and then delete all the CRD objects, and CRDs.
